File: NeverStand/NeverStand/CarWeightModels.swift
import Foundation

// Base data structure for car weight information
struct CarWeightData: Codable {
    let trainNumber: String
    let cn1: String
    let cid: String
    let stationID: String
    let cart1L: String
    let cart2L: String
    let cart3L: String
    let cart4L: String
    let cart5L: String
    let cart6L: String
    let utime: String
    
    enum CodingKeys: String, CodingKey {
        case trainNumber = "TrainNumber"
        case cn1 = "CN1"
        case cid = "CID"
        case stationID = "StationID"
        case cart1L = "Cart1L"
        case cart2L = "Cart2L"
        case cart3L = "Cart3L"
        case cart4L = "Cart4L"
        case cart5L = "Cart5L"
        case cart6L = "Cart6L"
        case utime
    }
}

// Structure for ranked car weight data
struct RankedCarWeight {
    let originalData: CarWeightData
    let rankedCarts: [Int: Int] // Key is cart index (1-6), value is rank (1-6)
    let selectedCar: Int        // Selected car (1-6)
}

File: NeverStand/NeverStand-Android/app/src/main/java/android_neverstand/model/CarWeightModels.kt
package android_neverstand.model

import com.google.gson.annotations.SerializedName

data class CarWeightData(
    @SerializedName("TrainNumber") val trainNumber: String,
    @SerializedName("CN1") val cn1: String,
    @SerializedName("CID") val cid: String,
    @SerializedName("StationID") val stationID: String,
    @SerializedName("Cart1L") val cart1L: String,
    @SerializedName("Cart2L") val cart2L: String,
    @SerializedName("Cart3L") val cart3L: String,
    @SerializedName("Cart4L") val cart4L: String,
    @SerializedName("Cart5L") val cart5L: String,
    @SerializedName("Cart6L") val cart6L: String,
    val utime: String
)

data class RankedCarWeight(
    val originalData: CarWeightData,
    val rankedCarts: Map<Int, Int>, // Key is cart index (1-6), value is rank (1-6)
    val selectedCar: Int        // Selected car (1-6)
)
